corehq/apps/linked_domain/tests/test_remote_auth.py

```diff
@@ -13,46 +12,62 @@
 
 
 class RemoteAuthTest(TestCase):
-    @classmethod
-    def setUpClass(cls):
-        super(RemoteAuthTest, cls).setUpClass()
 
-        cls.master_domain = uuid.uuid4().hex
-        cls.linked_domain = uuid.uuid4().hex
+    def test_returns_401_if_no_headers(self):
+        resp = self.client.get(self.url)
+        self.assertEqual(resp.status_code, 401)
 
-        cls.domain = create_domain(cls.master_domain)
-        cls.couch_user = WebUser.create(cls.master_domain, "test", "foobar", None, None)
-        cls.django_user = cls.couch_user.get_django_user()
-        cls.api_key, _ = HQApiKey.objects.get_or_create(user=cls.django_user)
+    def test_returns_400_if_no_remote_header_included(self):
+        headers = {'HTTP_AUTHORIZATION': f'apikey test:{self.api_key.key}'}
+        resp = self.client.get(self.url, **headers)
 
-        cls.auth_headers = {'HTTP_AUTHORIZATION': 'apikey test:%s' % cls.api_key.key}
+        self.assertEqual(resp.status_code, 400)
 
-        cls.linked_domain_requester = absolute_reverse('domain_homepage', args=[cls.linked_domain])
-        cls.domain_link = DomainLink.link_domains(cls.linked_domain_requester, cls.master_domain)
+    def test_returns_403_if_remote_header_specifies_incorrect_requester(self):
+        headers = {
+            'HTTP_AUTHORIZATION': f'apikey test:{self.api_key.key}',
+            REMOTE_REQUESTER_HEADER: 'wrong',
+        }
+        resp = self.client.get(self.url, **headers)
 
-    @classmethod
-    def tearDownClass(cls):
-        cls.couch_user.delete(cls.domain.name, deleted_by=None)
-        cls.api_key.delete()
-        cls.domain_link.delete()
-        cls.domain.delete()
-        super(RemoteAuthTest, cls).tearDownClass()
-
-    def test_remote_auth(self):
-        url = reverse('linked_domain:toggles', args=[self.master_domain])
-        resp = self.client.get(url)
-        self.assertEqual(resp.status_code, 401)
+        self.assertEqual(resp.status_code, 403)
 
-        resp = self.client.get(url, **self.auth_headers)
-        self.assertEqual(resp.status_code, 400)
+    def test_returns_403_if_no_linked_domain_access(self):
+        headers = {
+            'HTTP_AUTHORIZATION': f'apikey test:{self.api_key.key}',
+            REMOTE_REQUESTER_HEADER: self.downstream_domain_requester,
+        }
+        with mock.patch('corehq.apps.linked_domain.decorators.can_user_access_linked_domains', return_value=False):
```

Comment:
Interesting I didn't think of specifying the module as the object when using `patch.object`. I like that. I think my personal preference is to avoid specifying patches in setup that only apply to a subset of tests in a class, so in this case I prefer using `patch.object` over a move to setup. I'll stick with that unless you feel strongly the other way.